### ISSUE TITLE
Cannot handle other associative types

### ISSUE THREAD

**Issue by cstjean** [JuliaCollections/Memoize.jl#33]
```julia
julia> const MyDict = Dict
Dict

julia> @memoize MyDict f(x) = x+2
ERROR: LoadError: UndefVarError: MyDict not defined
Stacktrace:
 [1] top-level scope at none:0
 [2] eval(::Module, ::Any) at ./boot.jl:319
 [3] @memoize(::LineNumberNode, ::Module, ::Vararg{Any,N} where N) at /home/cst-jean/.julia/dev/Memoize/src/Memoize.jl:96
in expression starting at REPL[8]:1
```

This is because the dictionary expression is `eval`'ed in `Memoize`. It precludes using any associative type not defined in base.

... could you please pick a maintainer for this package? It's super useful, and it'd be great to fix the last few pain points.

**Comment by kmsquire** [comment]
Pretty much the same issue as #32.

**Comment by kmsquire** [comment]
This was not actually fixed by #46--even `eval`ing in `Main` doesn't mean that the corresponding dictionary is defined.
